AppCenter/AppCenter/Internals/HttpClient/MSHttpClient.m

```diff
@@ -2,20 +2,50 @@
 // Licensed under the MIT License.
 
 #import "MSHttpClient.h"
+#import "MSAppCenterErrors.h"
+#import "MSAppCenterInternal.h"
+#import "MSConstants+Internal.h"
+#import "MSHttpCall.h"
+#import "MSHttpClientPrivate.h"
+#import "MSHttpUtil.h"
+#import "MSLoggerInternal.h"
+#import "MSUtility+StringFormatting.h"
+#import "MS_Reachability.h"
 
 @implementation MSHttpClient
 
-- (instancetype)init {
+#define DEFAULT_RETRY_INTERVALS @[ @10, @(5 * 60), @(20 * 60) ]
 
-  // TODO implement this properly.
-  return self = [super init];
+- (instancetype)init {
+  return [self initWithMaxHttpConnectionsPerHost:nil
+                                  retryIntervals:DEFAULT_RETRY_INTERVALS
+                                    reachability:[MS_Reachability reachabilityForInternetConnection]];
 }
 
 - (instancetype)initWithMaxHttpConnectionsPerHost:(int)maxHttpConnectionsPerHost {
```

Comment:
Why the choice of `int` here vs `NSInteger` before?

Reply:
Do you mean because the next init uses `NSNumber`? That is because we need it to be nullable.

Reply:
The default value is different on iOS (4) and macos (6), so we want the type to be nullable to use the default. The NSNumber constructor is not public.